compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/declarations/IrDeclarations.kt

```diff
@@ -20,10 +23,13 @@ fun <D : IrAttributeContainer> D.copyAttributes(other: IrAttributeContainer?): D
 }
 
 val IrClass.isSingleFieldValueClass: Boolean
-    get() = valueClassRepresentation is InlineClassRepresentation
+    get() = this.isValue && valueClassRepresentation is InlineClassRepresentation
+
+val IrClass.isInline: Boolean
+    get() = (isSingleFieldValueClass && annotations.hasAnnotation(JVM_INLINE_ANNOTATION_FQ_NAME)) || (isValue && modality == Modality.SEALED)
```

Comment:
It almost seems like it would be a good idea to add another subclass of `ValueClassRepresentation`, `SealedInlineClassRepresentation`, for example to keep the invariant `isValue == (valueClassRepresentation != null)`

Reply:
Introduced `SealedInlineClassRepresentation`, and used it everywhere.